mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp

```diff
@@ -66,18 +71,95 @@ class SYCLInlinerInterface : public mlir::DialectInlinerInterface {
   }
 };
 
+class SYCLOpAsmInterface final : public mlir::OpAsmDialectInterface {
+public:
+  using mlir::OpAsmDialectInterface::OpAsmDialectInterface;
+
+  AliasResult getAlias(mlir::Type Type, llvm::raw_ostream &OS) const final;
+
+private:
+  static constexpr llvm::StringRef
+  getAlias(mlir::sycl::MemoryAccessMode MemAccessMode) {
```

Comment:
Thanks! This will make the type aliases more compact and at the same type preserve the meaning.